health-services/project-factory/src/server/api/campaignApis.ts

```
@@ -1327,60 +1231,118 @@ async function performAndSaveResourceActivity(
+        if (isMicroplanSource && facilityDataForMicroplan) {
+          const rowNumber = data['!row#number!'];
+          const microplanFacilityData = facilityDataForMicroplan.find((microplanFacility: any) => microplanFacility['!row#number!'] == rowNumber) || null;
+          if (microplanFacilityData) {
+            microplanFacilityData.facilityDetails.id = id
+          }
+        }
+      }
+    }
+  }
+  request.body.sheetErrorDetails = request?.body?.sheetErrorDetails
+    ? [...request?.body?.sheetErrorDetails, ...errors]
```

Biome (1.9.4)

[error] Unsafe usage of optional chaining.

If it short-circuits with 'undefined' the evaluation will throw TypeError here:


(lint/correctness/noUnsafeOptionalChaining)